src/sqlfluff/dialects/dialect_athena.py

```diff
@@ -0,0 +1,541 @@
+    StoredByGrammar=Sequence(
+        "STORED",
+        "BY",
+        Ref("QuotedLiteralSegment"),
+        Ref("SerdePropertiesGrammar", optional=True),
+    ),
+    StorageFormatGrammar=OneOf(
+        Sequence(
+            Ref("RowFormatClauseSegment", optional=True),
+            Ref("StoredAsGrammar", optional=True),
+        ),
+        Ref("StoredByGrammar"),
+    ),
+    CommentGrammar=Sequence("COMMENT", Ref("QuotedLiteralSegment")),
+    PartitionSpecGrammar=Sequence(
+        "PARTITION",
+        Bracketed(
+            Delimited(
+                Sequence(
+                    Ref("ColumnReferenceSegment"),
+                    Sequence(
+                        Ref("EqualsSegment"),
+                        Ref("LiteralGrammar"),
+                        optional=True,
+                    ),
+                )
+            )
+        ),
+    ),
+    BackQuotedIdentifierSegment=NamedParser(
+        "back_quote",
+        CodeSegment,
+        name="quoted_identifier",
+        type="identifier",
+    ),
+)
+
+athena_dialect.replace(
+    QuotedLiteralSegment=OneOf(
+        NamedParser("single_quote", CodeSegment, name="quoted_literal", type="literal"),
+        NamedParser("double_quote", CodeSegment, name="quoted_literal", type="literal"),
+        NamedParser("back_quote", CodeSegment, name="quoted_literal", type="literal"),
+    ),
+    LiteralGrammar=OneOf(
+        Ref("QuotedLiteralSegment"),
+        Ref("NumericLiteralSegment"),
+        Ref("BooleanLiteralGrammar"),
+        Ref("QualifiedNumericLiteralSegment"),
+        # NB: Null is included in the literals, because it is a keyword which
+        # can otherwise be easily mistaken for an identifier.
+        Ref("NullLiteralSegment"),
+        Ref("DateTimeLiteralGrammar"),
+        Sequence(Ref("SimpleArrayTypeGrammar"), Ref("ArrayLiteralSegment")),
+    ),
+    SimpleArrayTypeGrammar=Ref.keyword("ARRAY"),
+    TrimParametersGrammar=Nothing(),
+    SingleIdentifierGrammar=ansi_dialect.get_grammar("SingleIdentifierGrammar").copy(
+        insert=[
+            Ref("BackQuotedIdentifierSegment"),
+        ]
+    ),
+    BinaryOperatorGrammar=OneOf(
+        Ref("ArithmeticBinaryOperatorGrammar"),
+        Ref("StringBinaryOperatorGrammar"),
+        Ref("BooleanBinaryOperatorGrammar"),
+        Ref("ComparisonOperatorGrammar"),
+        # Add arrow operators for functions (e.g. filter)
+        Ref("RightArrowOperator"),
+    ),
+)
+
+
+class PrimitiveTypeSegment(BaseSegment):
+    """Primitive data types.
+
+    Since DDL is based on Hive and DML based on Prestodb this class has
+    primitives that may not work on specific situations
+    - Hive: https://cwiki.apache.org/confluence/display/hive/languagemanual+types
+    - PrestoDb: https://prestodb.io/docs/0.217/language/types.html
+    """
+
+    type = "primitive_type"
+    match_grammar = OneOf(
+        "BOOLEAN",
+        "TINYINT",
+        "SMALLINT",
+        "INTEGER",
+        "INT",  # Hive only
```

Comment:
What do comments like this mean? Does this mean `INT` is not a valid Presto data type? If so, do we plan to remove things like this?

Reply:
Thanks for pointing this out. Both engines support `int`. 
Also removed inline comments.